### DISCUSSION TITLE
The ability to transfer xp from your xp bar/levels into experience bottles

### DISCUSSION THREAD

**Discussion by jakeyhanna** [PurpurMC/Purpur#812] [__Dismissed__] [answered]
The ability to make XP level into experience bottles which you can give to your friends to use!

**Comment by YouHaveTrouble** [comment]
How would that work exactly?
How would player get their xp into a bottle?

**Reply by jakeyhanna** [reply]
Using these equations:

`Total experience =
level^2 + 6 × level (at levels 0–16)
2.5 × level^2 – 40.5 × level + 360 (at levels 17–31)
4.5 × level^2 – 162.5 × level + 2220 (at levels 32+)`

We can determine how much xp is put into the bottle/ How many bottles are needed to be given

**Reply by YouHaveTrouble** [reply]
that's not the issue here. I'm asking how would player get experience into a bottle?

**Comment by BillyGalbreath** [comment, marked as answer]
This is something better handled by a plugin only because there is a million different ways to implement such a feature.

Things to keep in mind:
- Bottle of Enchanting don't _actually_ hold any experience amount. Instead, they randomly [give 3-11 when they are used](https://minecraft.fandom.com/wiki/Bottle_o%27_Enchanting#Usage) (not created). So, how much experience points do we pull from a player that's creating them? (that's a rhetorical question, everyone will have their own different answer)
- How will the player create a bottle of enchanting? Shift right click (similar to how I coded [this mending feature](https://github.com/PurpurMC/Purpur/blob/ver/1.18.1/patches/server/0231-Shift-right-click-to-use-exp-for-mending.patch))? Right click a certain block? A crafting recipe? (another rhetorical question, as everyone will have their own different answer)
- Probably more stuff to think about, but I'm a bit side tracked at the moment to think of them all ^_^
